src/nvim/api/extmark.c

```diff
@@ -596,7 +600,10 @@ Integer nvim_buf_set_extmark(Buffer buffer, Integer ns_id, Integer line, Integer
   bool ephemeral = false;
   OPTION_TO_BOOL(ephemeral, ephemeral, false);
 
-  if (line < 0 || line > buf->b_ml.ml_line_count) {
+  bool strict = true;
+  OPTION_TO_BOOL(strict, strict, true);
+
+  if (line < 0 || (line > buf->b_ml.ml_line_count && strict)) {
```

Comment:
Don't we want non-strict to allow negative lines too (same for columns)?

Reply:
I thought about this but decided not, up for discussion though. -1 is used as a sentinel value for end of column, so I think it would be "weird" to allow -2

Reply:
Ok. I guess there is always pcall.

Reply:
idk @bfredl WDYT?